### PR TITLE
Homework/week3

### PR DESCRIPTION
**Pull request by hnnhack**
Homework week3 part-1 for submission

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,17 +26,125 @@
...
     });
+    const divElement = createAndAppend('div', rootElement, {
+      class: 'container',
+      id: 'container',
+    });
+    const select = createAndAppend('select', navElement);
+    try {
+      const repositories = await fetchJSON(url);
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      repositories.forEach((elem, index) => {
```
Comment (remarcmij) [inline homework/index.js]: The parameter name `elem` is okay, but `repository` would be better.

File: homework-classes/App.js [hunk 2]
```diff
@@ -20,11 +20,27 @@ class App {
...
-
+    Util.createAndAppend('h2', header, {
+      class: 'heading-two',
+      text: 'HYF Repositories:',
+    });
+    const selectElement = Util.createAndAppend('select', header);
     try {
       const repos = await Util.fetchJSON(url);
       this.repos = repos.map(repo => new Repository(repo));
       // TODO: add your own code here
```
Comment (remarcmij) [inline homework-classes/App.js]: Remove any comments that have outlived their purpose.

File: homework-classes/App.js [hunk 3]
```diff
@@ -20,11 +20,27 @@ class App {
...
+
+      repos.sort((a, b) => a.name.localeCompare(b.name));
```
Comment (remarcmij) [inline homework-classes/App.js]: There is a discrepancy between what you sort and what you use to display the repo details: the select box shows the `alumni` repo, but the details are for the `AngularJS` repo.

image

File: homework-classes/Contributor.js [hunk 4]
```diff
@@ -14,6 +14,26 @@ class Contributor {
    */
   render(container) {
     // TODO: replace the next line with your code.
```
Comment (remarcmij) [inline homework-classes/Contributor.js]: Remove this comment.

**Comment by hnnhack** [comment]
Hi Jim!
Yeah, that was simple but I didn't look carefully.
I am very thankfully for you. Not only for reviews, your classes, topics and especially for your tricks.
Wish you the best with your life, career, jobs and etc.

**Comment by remarcmij** [comment]
Hi Niyas, thank you for expressing your appreciation.

**Pull request closed without merging**